authorizer-app-backend/src/main/java/org/radarbase/authorizer/service/OAuth1RestSourceAuthorizationService.kt

```diff
@@ -86,11 +86,32 @@ abstract class OAuth1RestSourceAuthorizationService(
         }
     }
 
-    override fun deRegisterUser(user: RestSourceUser): RestSourceUser {
-        revokeToken(user)
-        val userDTO = userMapper.fromEntity(user)
-        userDTO.isAuthorized = false
-        return userRepository.update(user, userDTO)
+    override fun revokeToken(externalId: String, sourceType: String, token: TokenDTO): Boolean {
+        val authConfig = configMap[sourceType]
+            ?: throw HttpBadRequestException("client-config-not-found",
+                "Cannot find client configurations for source-type ${sourceType}")
+
+        if(token.accessToken.isNullOrEmpty()) throw HttpBadRequestException("token-empty", "Token cannot be null or empty")
+        val req = createRequest("DELETE",
+            authConfig.deregistrationEndpoint!!,
+            RestOauth1AccessToken(token.accessToken, ""),
+            sourceType)
+
+        return httpClient.newCall(req).execute().use { response ->
+            when (response.code) {
+                200, 204 -> true
+                400, 401, 403 -> false
+                else -> throw HttpBadGatewayException("Cannot connect to ${authConfig.deregistrationEndpoint}: HTTP status ${response.code}")
+            }
+        }
+    }
+
+    override fun deRegisterUser(user: RestSourceUser) {
+        return userRepository.delete(user)
+    }
+
+    override fun deleteUser(user: RestSourceUser) {
+        logger.info("Deleting user...")
```

Comment:
I think I mentioned it before about adding a Revoke Token button in the front end. What if we only allow deletions when the user is already unauthorized? So the deregister endpoint will only mark as unauthorized and the user will have to manually delete the user in the front end?

Reply:
I guess we should try not to complicate this for the users, we can just revokeToken and immediately delete the user after that. Garmin will send a deregister request to our endpoint later but the user will not be present in the repo so we will just return 200. This will be fine I think?

Worst case the user was not revoked properly by Garmin, then it will send some further data for that user to push endpoint, which will then again find that user is unauthorized (since not in repo) and revoke token again, so trying the flow again.

Reply:
Oh ok I see. Thanks